polkadot/xcm/pallet-xcm/src/benchmarking.rs

```diff
@@ -16,68 +16,157 @@
 
 use super::*;
 use bounded_collections::{ConstU32, WeakBoundedVec};
-use frame_benchmarking::{benchmarks, BenchmarkError, BenchmarkResult};
-use frame_support::weights::Weight;
+use frame_benchmarking::{benchmarks, whitelisted_caller, BenchmarkError, BenchmarkResult};
+use frame_support::{traits::Currency, weights::Weight};
 use frame_system::RawOrigin;
 use sp_std::prelude::*;
 use xcm::{latest::prelude::*, v2};
 
 type RuntimeOrigin<T> = <T as frame_system::Config>::RuntimeOrigin;
 
+// existential deposit multiplier
+const ED_MULTIPLIER: u32 = 100;
+
+/// Pallet we're benchmarking here.
+pub struct Pallet<T: Config>(crate::Pallet<T>);
+
+/// Trait that must be implemented by runtime to be able to benchmark pallet properly.
+pub trait Config: crate::Config {
+	/// A `MultiLocation` that can be reached via `XcmRouter`. Used only in benchmarks.
+	///
+	/// If `None`, the benchmarks that depend on a reachable destination will be skipped.
+	fn reachable_dest() -> Option<MultiLocation> {
+		None
+	}
+
+	/// A `(MultiAssets, MultiLocation)` pair representing assets and the destination they can
+	/// be teleported to. Used only in benchmarks.
+	///
+	/// Implementation should also make sure `dest` is reachable/connected.
+	///
+	/// If `None`, the benchmarks that depend on this will be skipped.
+	fn teleportable_assets_and_dest() -> Option<(MultiAssets, MultiLocation)> {
+		None
+	}
+
+	/// A `(MultiAssets, MultiLocation)` pair representing assets and the destination they can
+	/// be reserve-transferred to. Used only in benchmarks.
+	///
+	/// Implementation should also make sure `dest` is reachable/connected.
+	///
+	/// If `None`, the benchmarks that depend on this will be skipped.
+	fn reserve_transferable_assets_and_dest() -> Option<(MultiAssets, MultiLocation)> {
+		None
+	}
+}
+
 benchmarks! {
+	where_clause {
+		where
+			T: pallet_balances::Config,
```

Comment:
I am not sure, but probably this could affect other parachains if they dont use `pallet_balances` (maybe there are parachains without `pallet_balances`)

Reply:
make could be possible to rewrite benchmarking with `fungible` traits for [`Config` above](https://github.com/paritytech/polkadot-sdk/pull/1672/files#diff-299b81a18ba51e2c2fdaa3b6c7c7dd99ced49c9c7b8772fa1225ebecced332e0R34) or somehting similar instead `pallet_balances`
```
 <pallet_balances::Pallet<T> as Currency<_>>::make_free_balance_be(
pallet_balances::Pallet::<T>::free_balance(&caller)
```

Reply:
anyway, we are removing `Currency` trait  `<pallet_balances::Pallet<T> as Currency<_>>`, 
but this could be done in follow up PR also

Reply:
yes, I initially tried using `pallet-assets` but found the problem that few chains that use `pallet-xcm` also have `pallet-assets` - that's why I switched to `pallet-balances` which I saw is ubiquitous.

for maximum flexibility we can use custom `benchmarking::Config` trait and allow each chain to use whatever assets pallet they want, but that introduces a lot of duplicated code and extra complexity/work in benchmarks impls of all runtimes, so I decided we should not do it as part of this PR - let's go with general usecase pallet-balances and tackle custom approach if/when ever needed.